ngraph/test/type_prop/acos.cpp

```diff
@@ -0,0 +1,54 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+TEST(type_prop, acos_basic_shape_inference)
+{
+    Shape data_shape{2, 2};
+    const auto param = make_shared<op::Parameter>(element::f32, data_shape);
+    const auto op = make_shared<op::Acos>(param);
+    ASSERT_EQ(op->get_shape(), (data_shape));
+    ASSERT_EQ(op->get_element_type(), element::f32);
+}
+
+TEST(type_prop, acos_incompatible_input_type)
```

Comment:
[Spec](https://docs.openvinotoolkit.org/latest/openvino_docs_ops_arithmetic_Acosh_3.html) is saying that it should work only for floating point types. So, please check also that it throws for integral types.

Reply:
Since this test will have two cases where input type is not compatible I would suggest to use try-catch instead of ASSERT_THROW, for a more specific exception capture.
Here an [example](https://github.com/openvinotoolkit/openvino/pull/3409/files#diff-e27087594fb96c35a849c7723ca27f4dbda78980485a6b340526e42fac6d6477R572)

Reply:
Since we found out that it has one incompatible type I will resolve that